docs/source/fetching/persisted-queries.mdx

```diff
@@ -0,0 +1,159 @@
+
+Persisted queries also requires you to create and link a PQL and configure your router. For more information on the other configuration aspects of persisted queries, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries).
+
+### 1. Configure code generation
+
+The code generation requirements for persisted queries and APQ are the same: your operations need a SHA256 hash identifier included in the generated operation objects. To configure code generation accordingly, make the following changes in your `ApolloCodegenConfiguration` JSON file:
+
+1. In the `output` section of your configuration, set the [`operationManifest`](./../code-generation/codegen-configuration#operation-manifest) with:
+- a `path` to output the manifest file
+- the `version`: either `persistedQueries` for persisted queries or `legacyAPQs` for APQs
+
+```json title="apollo-codegen-config.json"
+"output": {
+    "operationManifest" : {
+        "path" : "./generated/operationIdentifiers.json",
+        "version" : "persistedQueries"
+    }
+}
+```
+
+2. In the `options` section of your configuration, set the `operationDocumentFormat` array to `definition`, `operationId`, or both `definition` and `operationId`. This determines how to generate the operation documents for your generated operations.
+
+```json title="apollo-codegen-config.json"
+"options": {
+    "operationDocumentFormat" : [
+        "definition",
+        "operationId"
+    ]
+}
+```
+
+Once you've configured code generation, you generate an operation manifest in one of two ways:
+- using the standard [`generate`](./../code-generation/codegen-cli#generate) command 
+- using the [`generate-operation-manifest`](./../code-generation/codegen-cli#generate-operation-manifest) command; this generates the operation manifest without running full code generation
+
+The resulting operation manifest for `persistedQueries` looks something like this: 
+
+```json title="operationIdentifiers.json"
+{
+  "format": "apollo-persisted-query-manifest",
+  "version": 1,
+  "operations": [
+    {
+      "id": "e0321f6b438bb42c022f633d38c19549dea9a2d55c908f64c5c6cb8403442fef",
+      "body": "query GetItem { thing { __typename } }",
+      "name": "GetItem",
+      "type": "query"
+    }
+  ]
+}
+```
+
+The resulting operation manifest JSON for `legacyAPQ` looks something like this: 
+
+```json title="operationIdentifiers.json"
+{
+  "525cd90917f8945d8d6a68fde991aaf766ff7d7b227ab2cd8dc97b7959c8d0d6" : {
+    "name": "GetItem",
+    "source": "query GetItem { thing { __typename ...Name } } fragment Name on Item { __typename name }"
+  }
+}
+```
+
+To automatically update the manifest for each new app release, you can include the [`generate`](./../code-generation/codegen-cli#generate) command in your CI/CD pipeline.
+
+#### Publish manifests to the PQL
```

Comment:
I don't think this section should be part of this doc since it doesn't really pertain to the iOS portion. At most I think if we want to mention/reference the PQL we should link out to a separate docs page for that.

Reply:
Hmm, I think it's preferable to have one place where the publish instructions are referenced for ease of maintenance but it's a pretty small section so I'm not completely opposed to having it here.

Reply:
I wrestled with this too, since this isn't iOS specific. At the same time, if I was a client maintainer, I'd prefer to have everything in one place, which is why I decided to put it in. If others feel strongly about removing it, we can do that.

Reply:
> Is this being duplicated into the Kotlin docs too?
Yes! I'd like to have parity between all client docs on this.

BTW, it'll also be in the ["main PQ" docs](https://github.com/apollographql/docs/pull/496) regardless, just want to make it easier for a client dev by having everything here. Am considering making reusable stubs for any sections that will be duplicated.

Reply:
I'd like to keep this section as well. It would be nice if we could share this block across the different platforms documentation though.

I know we have a folder of `shared` `.md` files that can be placed in-line inside other documentation pages in the iOS repo. But could we share this across the different docs platforms? That way we guarantee we're all using the same verbiage and it can be kept up to date in a single place. @trevorblades Is that possible?

Reply:
Good question, @AnthonyMDev ! I didn't think about/realize that these shared `.md` were on per-repo basis.
Maybe easiest just to make a new React component with just text in the main docs repo? Let me know what you think @trevorblades

Reply:
https://github.com/apollographql/docs/pull/515 exposes all of these shared bits of MDX as components that you can use in your docs' content:

```mdx
- some
- markdown
- content

<ObtainApiKey />
```

If you want to see the list of components that can be used, you can check out the export names in the `src/content/shared/index.js` file in the docs repo.

Reply:
Yay, ty!!!! Will replace text sections in client docs with these tomorrow.